crates/sui-types/src/crypto.rs

```diff
@@ -704,6 +703,24 @@ impl Signature {
         value.write(&mut message);
         secret.sign(&message)
     }
+
+    pub fn new_temp(value: &[u8], secret: &dyn signature::Signer<Signature>) -> Signature {
+        secret.sign(value)
+    }
+
+    // pub fn new_secure<T>(
```

Comment:
new_temp will be replaced by new_secure. `Signature::new` only works on Signable trait struct that we eventually will not need.